File: MusicWeb.Server/MusicWeb.Models/Entities/Albums/Album.cs
using MusicWeb.Models.Entities.Artists;
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Entities.Posts;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class Album : BaseEntity
    {
        public string Name { get; set; }
        public DateTime ReleaseDate { get; set; }
        public string ImagePath { get; set; }

        public int ArtistId { get; set; }
        public virtual Artist Artist { get; set; }

        public int AlbumGenreId { get; set; }
        public virtual Genre AlbumGenre { get; set; }
        public double duration { get; set; }
        public string description { get; set; }
        public Boolean isConfirmed { get; set; }

        public virtual ICollection<Song> Songs{ get; set; }
        public virtual ICollection<AlbumReview> AlbumReviews{ get; set; }
        public virtual ICollection<UserFavoriteAlbum> UserFavoriteAlbums{ get; set; }
        public virtual ICollection<ArtistsOnTheAlbum> ArtistsOnTheAlbums{ get; set; }
        //public virtual ICollection<Post> Posts { get; set; }
    }
}

File: MusicWeb.Client/src/models/Album.js
export default class Album {
  id = null;
  imagePath = "";
  name = "";
  releaseDate = "";
  artistId = "";
  albumGenreId = "";
  description = "";
  duration = "";
  isConfirmed = "";
  artist = {};
  albumGenre = {};
  albumReviews = [];
  songs = [];
  rating = "";
  ratingsCount = "";
  favoriteCount = null;
  reviewsCount = null;
  artistName = "";

  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.imagePath = $data.imagePath;
      this.name = $data.name;
      this.releaseDate = $data.releaseDate;
      this.artistId = $data.artistId;
      this.albumGenreId = $data.albumGenreId;
      this.duration = $data.duration;
      this.description = $data.description;
      this.isConfirmed = $data.isConfirmed;
      this.songs = $data.songs ? $data.songs : [];
      this.albumReviews = $data.albumReviews ? $data.albumReviews : [];
      this.artist = $data.artist ? $data.artist : {};
      this.albumGenre = $data.albumGenre ? $data.albumGenre : {};
      this.rating = $data.rating ? $data.rating : 0;
      this.ratingsCount = $data.ratingsCount ? $data.ratingsCount : 0;
      this.favoriteCount = $data.favoriteCount ? $data.favoriteCount : 0;
      this.reviewsCount = $data.reviewsCount ? $data.reviewsCount : 0;
      this.artistName = $data.artistName;
    }
  }
}
